File: app/src/main/java/org/dhis2/data/schedulers/SchedulerModule.java
package org.dhis2.data.schedulers;

import androidx.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SchedulerModule {
    private final SchedulerProvider schedulerProvider;

    public SchedulerModule(@NonNull SchedulerProvider schedulerProvider) {
        this.schedulerProvider = schedulerProvider;
    }

    @Provides
    @Singleton
    SchedulerProvider schedulerProvider() {
        return schedulerProvider;
    }
}

File: app/src/main/java/org/dhis2/data/schedulers/SchedulerModule.kt
package org.dhis2.data.schedulers

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class SchedulerModule(private val schedulerProvider: SchedulerProvider) {

    @Provides
    @Singleton
    internal fun schedulerProvider(): SchedulerProvider {
        return schedulerProvider
    }
}
